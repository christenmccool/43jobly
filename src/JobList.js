import React from 'react';
import JobCard from './JobCard';

/** JobList component for Jobly app
 * Renders a list of JobCards given a jobs list prop
 */

const JobList = ({jobs, apply}) => {
  
  return (
    <div className="JobList">
      {jobs.map(job => 
        <JobCard 
          title={job.title} 
          salary={job.salary} 
          equity={job.equity} 
          id={job.id}
          key={job.id}
          companyName={job.companyName || null}
          companyHandle={job.companyHandle || null}
          apply={apply}
        />
      )}
    </div>
  )
}

export default JobList;