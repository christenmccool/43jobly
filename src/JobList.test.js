import { render, asFragment } from '@testing-library/react';
import JobList from './JobList';

test('if it renders without crashing', () => {
  render(<JobList jobs={[{title:"Test title 1", salary:100, equity:0.5},{title:"Test title 2", salary:200, equity:0.8}]}/>);
});

test('if it matches snapshot', () => {
  const {asFragment} = render(<JobList jobs={[{title:"Test title 1", salary:100, equity:0.5},{title:"Test title 2", salary:200, equity:0.8}]}/>);
  expect(asFragment()).toMatchSnapshot();
});

test('if it renders title, salary, and equity of jobs', () => {
  const {queryByText} = render(<JobList jobs={[{title:"Test title 1", salary:100, equity:0.5},{title:"Test title 2", salary:200, equity:0.8}]}/>);
  
  expect(queryByText("Test title 1")).toBeInTheDocument();
  expect(queryByText("Salary: 100")).toBeInTheDocument();
  expect(queryByText("Equity: 0.5")).toBeInTheDocument();
  
  expect(queryByText("Test title 2")).toBeInTheDocument();
  expect(queryByText("Salary: 200")).toBeInTheDocument();
  expect(queryByText("Equity: 0.8")).toBeInTheDocument();
});

