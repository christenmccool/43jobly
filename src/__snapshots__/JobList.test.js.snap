// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`if it matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="JobList"
  >
    <div
      class="JobCard"
    >
      <div>
        <h1
          class="JobCard-title"
        >
          Test title 1
        </h1>
        <button
          class="JobCard-btn JobCard-apply-btn"
        >
          Apply
        </button>
      </div>
      <p
        class="JobCard-info"
      >
        Salary: 100
      </p>
      <p
        class="JobCard-info"
      >
        Equity: 0.5
      </p>
    </div>
    <div
      class="JobCard"
    >
      <div>
        <h1
          class="JobCard-title"
        >
          Test title 2
        </h1>
        <button
          class="JobCard-btn JobCard-apply-btn"
        >
          Apply
        </button>
      </div>
      <p
        class="JobCard-info"
      >
        Salary: 200
      </p>
      <p
        class="JobCard-info"
      >
        Equity: 0.8
      </p>
    </div>
  </div>
</DocumentFragment>
`;
